src/rtctools_heat_network/heat_physics_mixin.py

```diff
@@ -1462,6 +1538,274 @@ def __pipe_heat_to_discharge_path_constraints(self, ensemble_member):
+                        np.inf,
+                    )
+                )
+                constraints.append(
+                    (
+                        ates_temperature_disc
+                        - sup_temperature_disc
+                        - is_buffer_charging * big_m,
+                        -np.inf,
+                        0.0,
+                    )
+                )
+                # inequality constraint when charging, carrier temperature should be larger or equal to ates temperature
+                constraints.append(
+                    (
+                        sup_temperature_disc
+                        - ates_temperature
+                        + (1-is_buffer_charging) * big_m,
+                        0.0,
+                        np.inf,
+                    )
+                )
+
+        return constraints
+
+    def __get_linear_Tloss_vs_storedheat(self, heat_max, temperature_ates, temperature_ambient=17, n_lines=5):
+        """
+        Function to linearise the temperature loss based on:
+        - the current discrete temperature of the ATES well
+        - the current ambient temperature
+        - the current stored heat
+        The current stored heat is split in segments of equal size to linearise into n_lines
+        numbers of lines
+        Tloss/dt = a*stored_heat/heat_max + b
+        """
+
+        heat_points = np.linspace(1, heat_max, n_lines+1)/heat_max # cannot be 0
+
+        def algebraic_Tloss_ates(heat_factor, heat_max, temperature_ates, temperature_ambient):
+            #TODO: function needs to be updated with realistic function
+            dTdt = 1e-7*(temperature_ates-temperature_ambient)*np.exp(-heat_factor)
+            return dTdt
+
+        Tloss_dt_points = np.array([algebraic_Tloss_ates(h, heat_max, temperature_ates, temperature_ambient) for h in heat_points])
+
+        a = np.diff(Tloss_dt_points) / np.diff(heat_points)
+        b = Tloss_dt_points[1:]-a*heat_points[1:]
+
+        return a, b
+
+    def __get_linear_heatloss_vs_storedheat(self, heat_max, temperature_ates, temperature_ambient=17, n_lines=5):
+        """
+        Function to linearise the temperature loss based on:
+        - the current discrete temperature of the ATES well
+        - the current ambient temperature
+        - the current stored heat
+        The current stored heat is split in segments of equal size to linearise into n_lines
+        numbers of lines
+        heatloss/dt = a*stored_heat + b
+        """
+        #TODO: ensure heatloss and temperature loss linearisation rely on the same equations
+        heat_points = np.linspace(1.0, heat_max, n_lines+1) # cannot be 0
+
+        def algebraic_heatloss_ates(heat_points, heat_max, temperature_ates, temperature_ambient):
+            #TODO: function needs to be updated with realistic function, woudl normally non convex
+            heatloss = 1e-10*heat_points * (temperature_ates-temperature_ambient)*np.exp(-heat_points/heat_max)
```

Comment:
I don't get this. I would expect a quadratic relation w.r.t. the stored_heat here, as the size of the bell increases the energy losses increase quadratically as the surface area of the bell increases quadratically.

Reply:
In reality it is also not a quadratic function:

in case of a sphere:
heat loss scales scales with the surface area (A ~r^2)
stored heat scales with the volume (V ~r^3)
thus heat loss ~V^(2/3) ~ stored heat ^(2/3)

however assuming it will be a cylinder and heat is only lossed on the sides, not to the above/below clay layers (assuming height constant):
heat loss scales scales with the surface area sides (A ~2pi r * h)
stored heat scales with the volume (V ~pi r^2 * h)
thus heat loss ~V^(1/2) ~ stored heat ^(1/2)

in case of a cylinder with heat losses on top and bottom dominant(assuming h constant):
heat loss scales scales with the surface area sides (A ~ 2pi r^2)
stored heat scales with the volume (V ~pi r^2 * h)
thus heat loss ~ V

in case of a cylinder with heat losses on all sides (assuming h constant):
heat loss scales scales with the surface area sides (A ~2pi r * h + 2pi r^2)
stored heat scales with the volume (V ~pi r^2 * h)
thus heat loss ~ (V + V^(1/2)) 

I'll check which function is most relevant, for now I'll keep an efficiency that changes with the temperature and will be multiplied with the stored heat